web/profiles/joinup/src/Form/UserCancelForm.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace Drupal\joinup\Form;
+
+use Drupal\Component\Datetime\TimeInterface;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityManagerInterface;
+use Drupal\Core\Entity\EntityTypeBundleInfoInterface;
+use Drupal\Core\Form\ConfirmFormHelper;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\joinup_core\JoinupRelationManager;
+use Drupal\user\Form\UserCancelForm as CoreUserCancelForm;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Overrides the default implementation of the form to delete a user account.
+ *
+ * Deletion of a user account will be denied when the user is the sole owner of
+ * a collection.
+ */
+class UserCancelForm extends CoreUserCancelForm {
+
+  /**
+   * The relation manager service.
+   *
+   * @var \Drupal\joinup_core\JoinupRelationManager
+   */
+  protected $relationManager;
+
+  /**
+   * Instantiates a new UserCancelForm class.
+   *
+   * @param \Drupal\Core\Entity\EntityManagerInterface $entity_manager
+   *   The entity manager.
+   * @param \Drupal\Core\Entity\EntityTypeBundleInfoInterface $entity_type_bundle_info
+   *   The entity type bundle service.
+   * @param \Drupal\Component\Datetime\TimeInterface $time
+   *   The time service.
+   * @param \Drupal\joinup_core\JoinupRelationManager $relation_manager
+   *   The Joinup relation manager.
+   */
+  public function __construct(EntityManagerInterface $entity_manager, EntityTypeBundleInfoInterface $entity_type_bundle_info = NULL, TimeInterface $time = NULL, JoinupRelationManager $relation_manager = NULL) {
+    parent::__construct($entity_manager, $entity_type_bundle_info, $time);
+
+    // Replicate the behaviour of the parent implementation.
+    $this->relationManager = $relation_manager ?: \Drupal::service('joinup_core.relations_manager');
```

Comment:
Yeah but that's part of core and they do it because they cannot break backwards compatibility with earlier versions of Drupal 8 that had fewer arguments to this constructor.

Our code is new so we don't break backwards compatibility and don't need this workaround.